AK/HashTable.h

```diff
@@ -345,12 +447,14 @@ class HashTable {
     [[nodiscard]] size_t used_bucket_count() const { return m_size + m_deleted_count; }
     [[nodiscard]] bool should_grow() const { return ((used_bucket_count() + 1) * 100) >= (m_capacity * load_factor_in_percent); }
 
-    Bucket* m_buckets { nullptr };
+    BucketType* m_buckets { nullptr };
+
+    [[no_unique_address]] HouseKeepingType m_housekeeping;
```

Comment:
"House keeping" is not an algorithmic concept I'm familiar with. Unless this is an established term of art that I'm unaware of, we should either give this a better name, or at least explain what it is.

Reply:
So housekeeping generally means taking care of something or cleaning something up,
in this case I am keeping track of the first and last insertion and take care of their integrity

I genually can't really find a better name, and to be honest, it would be best to get rid of this struct, but asfar as I know you are not allowed to mark members as `requires(IsOrdered)`

Reply:
How about `OrderedCollectionData`?

Reply:
lets take that for now